### PR TITLE
No more infinite free slabs

### LINKED ISSUE #7107: IPCs can use lipid extractors.
## Round ID:
38875

## Testmerges:
No testmerges I think

## Reproduction:
Be IPC with max charge -> use lipid extractor -> Lipid extractor produces meat despite you being made of metal

### LINKED ISSUE #7123: Endless Meat
Issue reported from  Round ID: 38927 (BeeStation Sage - Roleplay)

Reporting client version: 514.1583

As human, became full on food. Drank a potion and became a skeleton, carrying over full status. Step inside lipid extractor and encounter endless extraction. Exiting at will is available, but after exiting full status will still be maintained. Cycle is able to be repeated since full status does not go away.

### PR DESCRIPTION
**Pull request by Dejaku51**
## About The Pull Request
Fixes #7107 
FIxes #7123

- Makes it so mobs that use electricity instead of nutrients (IPC, ethereals) can't use lipid extractor
- Created a failsafe in case admin tries to tweak component rating of the lipid extractor high enough that it uses negative amount of nutrients to create a slab by exploding the machine (server crashing adminbus bad)

## Why It's Good For The Game
Infinite free slabs bad. Fixes good
Server crashing bad. failsafes good

## Testing Photographs and Procedure

1. Be IPC/ethereal
2. Try to enter the lipid extractor
3. You can't
4. Be a fat human
5. Try to enter the lipid extractor
6. machine sucks the fat out of you

⁠
1. Tweak component rating of lipid extractor to 20
2. Insert fat human inside
3. Machine will explode

## Changelog
:cl:
fix: IPCs and ethereals now can't use the lipid extractor
admin: created a failsafe to avoid admins crashing the server using lipid extractor
/:cl:

### CODE REVIEW

File: code/game/machinery/fat_sucker.dm [hunk 1]
```diff
@@ -182,6 +182,14 @@
 	if(occupant && iscarbon(occupant))
 		var/mob/living/carbon/C = occupant
 		if(C.type_of_meat)
+			// Someone changed component rating high enough so it requires negative amount of nutrients to create a meat slab
+			if(nutrient_to_meat < 0)
+				message_admins("[src] requires negative amount of nutrients to create a slab. Exploding it to avoid crash") // no need to actually explode it but i want to punish whoever tries to do it
```
Comment (Archanial) [inline code/game/machinery/fat_sucker.dm]: Don't message admins this, since it's not possible to be achieved without admin powers

Comment (Dejaku51) [reply code/game/machinery/fat_sucker.dm]: done

**Comment by EvilDragonfiend** [comment]
I don't get the explosion mechanism.... wouldn't you just put a maximum there?
`nutrient_to_meat = max(initial(nutrient_to_meat) - nutriment_rating * 5, 5)`

**Review by Archanial — DISMISSED** [review]

**Comment by Dejaku51** [comment]
> I don't get the explosion mechanism.... wouldn't you just put a maximum there? `nutrient_to_meat = max(initial(nutrient_to_meat) - nutriment_rating * 5, 5)`

i could but i like explosions and i want to punish whoever almost crashed the game :trolled:

**Comment by Dejaku51** [comment]
made it so only organics can use it (see #7123 )

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**